src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryEvaluator.cs

```diff
@@ -0,0 +1,119 @@
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private string GetField(FieldExpression f)
+        {
+            return f.FieldValue.Length != 0 ? f.FieldValue : f.FieldValueWithoutAlias;
+        }
+
+        private IQueryMatch Evaluate(QueryExpression where)
+        {
+            switch (@where)
+            {
+                case MethodExpression me:
+                    var expressionType = QueryMethod.GetMethodType(me.Name.Value);
+                    switch (expressionType)
+                    {
+                        case MethodType.StartsWith:
+                            return _searcher.StartWithQuery(GetField((FieldExpression)me.Arguments[0]), ((ValueExpression)me.Arguments[1]).Token.Value);
+                        default:
+                            return null;
+                    }
+                case TrueExpression _:
+                case null:
+                    return null;
+                case InExpression ie:
+                    return (ie.Source, ie.Values) switch
+                    {
+                        (FieldExpression f, List<QueryExpression> list) => EvaluateInExpression(f, list),
+                        _ => throw new NotSupportedException()
+                    };
+                case BinaryExpression be:
+                    return (be.Operator, be.Left, be.Right) switch
+                    {
+                        (OperatorType.Equal, FieldExpression f, ValueExpression v) => _searcher.TermQuery(GetField(f), v.Token.Value),
+                        (OperatorType.And, QueryExpression q1, QueryExpression q2) => _searcher.And(Evaluate(q1), Evaluate(q2)),
+                        (OperatorType.Or, QueryExpression q1, QueryExpression q2) => _searcher.Or(Evaluate(q1), Evaluate(q2)),
+                        _ => throw new NotSupportedException()
+                    };
+                default:
+                    return null;
+            }
+        }
+
+        private IQueryMatch EvaluateInExpression(FieldExpression f, List<QueryExpression> list)
+        {
+            var values = new List<string>();
+            foreach (ValueExpression v in list)
+                values.Add(v.Token.Value);
+
+            return _searcher.InQuery(f.FieldValue, values);
+        }
+
+        private MatchCompareFieldType OrderTypeFieldConventer(OrderByFieldType original) =>
+            original switch
+            {
+                OrderByFieldType.Double => MatchCompareFieldType.Floating,
+                OrderByFieldType.Long => MatchCompareFieldType.Integer,
+                OrderByFieldType.AlphaNumeric => MatchCompareFieldType.Integer,
+                _ => MatchCompareFieldType.Sequence
+            };
+        
+        private IQueryMatch OrderByEvaluate(IQueryMatch result, FieldsToFetch fieldsToFetch, List<(QueryExpression Expression, OrderByFieldType FieldType, bool Ascending)> orderList)
+        {
+            foreach (var order in orderList)
```

Comment:
Pretty sure that this is wrong. We need to have a compound ordering.
We can't nest sorts

Reply:
I don't know what you mean. Can you explain?

Reply:
`order by Name, Age` means sort first by Name, then if the name is equal, by Age.

Your code is sort by name, then sort by age (equivalent to sorting by Age only)

Reply:
This has to be implemented as a Corax primitive.